File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string ans = "";
        for(char &ch : s) {
            if(ch == '*') {
                if(!ans.empty()) {
                    ans.pop_back();
                } else {
                    continue;
                }
            } else if(ch == '#') {
                ans = ans + ans;
            } else if(ch == '%') {
                reverse(ans.begin(), ans.end());
            } else {
                ans += ch;
            }
        }
        return ans;
    }
};

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        ans = []
        for ch in s:
            if ch == '*':
                if ans:
                    ans.pop()
            elif ch == '#':
                ans.extend(ans)
            elif ch == '%':
                ans.reverse()
            else:
                ans.append(ch)
        return "".join(ans)